### PR TITLE
vSphere Destroy: add log messages for tag destruction

### PR DESCRIPTION
**Pull request by patrickdillon**
Prints Info message when tag and tag categories are successfully destroyed.

/test e2e-vsphere

### CODE REVIEW

File: pkg/destroy/vsphere/vsphere.go [hunk 1]
```diff
@@ -217,12 +217,14 @@ func (o *ClusterUninstaller) Run() error {
...
 		return err
 	}
+	o.Logger.WithField("Tag", o.InfraID).Info("Destroyed")
 
 	o.Logger.Debug("Delete tag category")
 	if err = deleteTagCategory(context.TODO(), o.RestClient, "openshift-"+o.InfraID); err != nil {
 		o.Logger.Errorln(err)
 		return err
 	}
+	o.Logger.WithField("Tag Category", "openshift-"+o.InfraID).Info("Destroyed")
```
Comment (staebler) [inline pkg/destroy/vsphere/vsphere.go:227]: It is a bit easier to read log entries when the log fields do not have spaces.
```suggestion
	o.Logger.WithField("TagCategory", "openshift-"+o.InfraID).Info("Destroyed")
```

File: pkg/destroy/vsphere/vsphere.go [hunk 2]
```diff
@@ -217,12 +217,14 @@ func (o *ClusterUninstaller) Run() error {
 		o.Logger.Errorln(err)
 		return err
 	}
+	o.Logger.WithField("Tag", o.InfraID).Info("Destroyed")
```
Comment (staebler) [inline pkg/destroy/vsphere/vsphere.go:220]: For consistency, if we are going to add the field to the "Destroyed" log event, we should add it to the other log events for the action. Specifically, use the logger returned by the call to `WithField` for the "Delete tag" log event and the error log event as well.

Comment (patrickdillon) [reply pkg/destroy/vsphere/vsphere.go]: Ok. I think the new changes are in line with this suggestion.

**Review by staebler — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**